### PR TITLE
0.6.1 history class

### PR DESCRIPTION
**Pull request by ianjennings**
- New search class (previously history)
- ```.history()``` -> ```.search()```
- Search filters can be added with plugins

History now works like this:

```js
    chatHistory.search({
            event: 'tester',
            limit: 50
        }).on('tester', (a) => {

            assert.equal(a.event, 'tester');

            count += 1;

        }).on('$.search.finish', () => {
            assert.equal(count, 50, 'correct # of results');
            done();
        });
```

### CODE REVIEW

File: src/bootstrap.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ module.exports = (ceConfig, pnConfig) => {
     };
 
     ChatEngine.protoPlugins = {};
```
Comment (raimondi1337) [inline src/bootstrap.js]: Should this be renamed too?

File: test/integration/main.test.js [hunk 2]
```diff
@@ -166,40 +166,37 @@ describe('history', () => {
...
 
-        let chatHistory = new ChatEngine.Chat('chat-history-2', false);
-
-        chatHistory.on('$.history.tester', (a) => {
-
-            assert.equal(a.event, 'tester');
+        chatHistory = new ChatEngine.Chat('chat-history-8', false);
 
-            count += 1;
+        // let i = 0;
```
Comment (raimondi1337) [inline test/integration/main.test.js]: Is there a reason to keep these comment blocks around?

File: test/integration/main.test.js [hunk 3]
```diff
@@ -211,39 +208,47 @@ describe('history', () => {
 
         let chatHistory2 = new ChatEngine.Chat('chat-history-3', false);
 
-        chatHistory2.on('$.history.tester', (a) => {
+        // let i = 1;
```
Comment (raimondi1337) [inline test/integration/main.test.js]: Is there a reason to keep these comment blocks around?

File: src/bootstrap.js (hunk 1 above)
Comment (ianjennings) [reply src/bootstrap.js]: the proto stuff is a separate pull

Comment (ianjennings) [reply src/bootstrap.js]: but yeah, maybe we can call them protos.

File: test/integration/main.test.js (hunk 2 above)
Comment (ianjennings) [reply test/integration/main.test.js]: That's a good question. I feel like I have history set to expire in 1 day or something on this key. let me undo that.

**Pull request merged**